juniper_actix_web/src/lib.rs

```diff
@@ -0,0 +1,620 @@
+use futures::{future, Future};
+
+use actix_web::{
+    dev,
+    http::{Method, StatusCode},
+    web, Error, FromRequest, HttpRequest, HttpResponse, Responder,
+};
+
+use juniper::http::{
+    graphiql, playground, GraphQLRequest as JuniperGraphQLRequest,
+    GraphQLResponse as JuniperGraphQLResponse,
+};
+
+use juniper::{
+    serde::Deserialize, DefaultScalarValue, FieldError, GraphQLType, InputValue, RootNode,
+    ScalarRefValue, ScalarValue,
+};
+
+use serde::{de, Deserializer};
+
+fn deserialize_non_empty_vec<'de, D, T>(deserializer: D) -> Result<Vec<T>, D::Error>
+where
+    D: Deserializer<'de>,
+    T: Deserialize<'de>,
+{
+    let v = Vec::<T>::deserialize(deserializer)?;
+
+    if v.is_empty() {
+        Err(de::Error::invalid_length(0, &"a positive integer"))
+    } else {
+        Ok(v)
+    }
+}
+
+#[derive(Debug, serde_derive::Deserialize, PartialEq)]
+#[serde(untagged)]
+#[serde(bound = "InputValue<S>: Deserialize<'de>")]
+enum GraphQLBatchRequest<S = DefaultScalarValue>
```

Comment:
We really need to factor out the batch request into the main crate, right now we are just duplicating it everywhere. 

(You would be very welcome to do this if interested, but just a suggestion. We can do this as a follow up. )

Reply:
Good to know—should I leave this in here for now then?

Reply:
Good to know—once I wrap this up I'd definitely be interested in taking a look